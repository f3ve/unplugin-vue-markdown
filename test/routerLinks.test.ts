import { describe, expect, it } from 'vitest'
import { resolveOptions } from '../src/core/options'
import { createMarkdown } from '../src/core/markdown'

describe('router link transforms', async () => {
  const optionsNoLink = resolveOptions({})
  const optionsWithCustomLink = resolveOptions({
    routerLinkComponent: 'MyCustomRouterLink',
  })
  const options = resolveOptions({
    routerLinkComponent: 'RouterLink',
  })

  const markdownToVue = await createMarkdown(options)
  const mdWithoutRouterLink = await createMarkdown(optionsNoLink)
  const mdWithCustomLink = await createMarkdown(optionsWithCustomLink)

  it('transforms internal links to router links', () => {
    const md = `
# Internal links to Router links!

[test](/)
[test 1234](/test/1234)
[test slug](/blog/this-is-my-test-slug-12345)
    `

    expect(markdownToVue('', md).code).toMatchSnapshot()
  })

  it('does not transform non-internal links to router links', () => {
    const md = `
# Non-internal links

[test](https://example.com)
[test 1234](#header1234)
[test url with long path](https://example.com/this/is/a/really/long/path)
    `
    expect(markdownToVue('', md).code).toMatchSnapshot()
    expect(mdWithCustomLink('', md).code).toMatchSnapshot()
  })

  it ('supports custom router link components', () => {
    const md = `
# Internal links to Router links!

[test](/)
[test 1234](/test/1234)
[test slug](/blog/this-is-my-test-slug-12345)
    `
    expect(mdWithCustomLink('', md).code).toMatchSnapshot()
  })

  it('does not convert to router links if router link option is not provided', () => {
    const md = `
# Internal links to Router links!

[test](/)
[test 1234](/test/1234)
[test slug](/blog/this-is-my-test-slug-12345)
    `
    expect(mdWithoutRouterLink('', md)).toMatchSnapshot()
  })
})
