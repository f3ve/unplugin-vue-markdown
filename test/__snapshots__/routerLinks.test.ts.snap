// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`router link transforms > does not convert to router links if router link option is not provided 1`] = `
{
  "code": "<template><div class="markdown-body"><h1>Internal links to Router links!</h1>
<p><a href="/">test</a>
<a href="/test/1234">test 1234</a>
<a href="/blog/this-is-my-test-slug-12345">test slug</a></p>
</div></template>
<script setup>
const frontmatter = {}
defineExpose({ frontmatter })
</script>",
  "map": {
    "mappings": "",
  },
}
`;

exports[`router link transforms > does not transform non-internal links to router links 1`] = `
"<template><div class="markdown-body"><h1>Non-internal links</h1>
<p><a href="https://example.com">test</a>
<a href="#header1234">test 1234</a>
<a href="https://example.com/this/is/a/really/long/path">test url with long path</a></p>
</div></template>
<script setup>
const frontmatter = {}
defineExpose({ frontmatter })
</script>"
`;

exports[`router link transforms > does not transform non-internal links to router links 2`] = `
"<template><div class="markdown-body"><h1>Non-internal links</h1>
<p><a href="https://example.com">test</a>
<a href="#header1234">test 1234</a>
<a href="https://example.com/this/is/a/really/long/path">test url with long path</a></p>
</div></template>
<script setup>
const frontmatter = {}
defineExpose({ frontmatter })
</script>"
`;

exports[`router link transforms > supports custom router link components 1`] = `
"<template><div class="markdown-body"><h1>Internal links to Router links!</h1>
<MyCustomRouterLink to="/">test</MyCustomRouterLink><MyCustomRouterLink to="/test/1234">test 1234</MyCustomRouterLink><MyCustomRouterLink to="/blog/this-is-my-test-slug-12345">test slug</MyCustomRouterLink></div></template>
<script setup>
const frontmatter = {}
defineExpose({ frontmatter })
</script>"
`;

exports[`router link transforms > transforms internal links to router links 1`] = `
"<template><div class="markdown-body"><h1>Internal links to Router links!</h1>
<RouterLink to="/">test</RouterLink><RouterLink to="/test/1234">test 1234</RouterLink><RouterLink to="/blog/this-is-my-test-slug-12345">test slug</RouterLink></div></template>
<script setup>
const frontmatter = {}
defineExpose({ frontmatter })
</script>"
`;
